src/main/scala/asciibot/parser/parser.scala

```diff
@@ -0,0 +1,200 @@
+    }
+    while (lines(0)(e - 1) == ' ' && lines(1)(e - 1) == ' ' && lines(1)(e - 1) == ' ') {
+      e -= 1;
+    }
+    (s, e)
+  }
+
+  def parseInState(a: Seq[String], start: Int, end: Int) : (Surroundings, State) = {
+    def throwBE(m : String) = throw new CharBlockError(m, start, end)
+    assert(a.length == 3)
+    val state_str = a(1).slice(start+1,end-1)
+    // Check to make sure there are no illegal characters in the state name
+    if ("#*_|-/> " exists (state_str contains _)) {
+      //throw new StateNameError(state_str)
+      throw new CharBlockError(s"Bad state name `$state_str'", start, end)
+    }
+
+    if (   a(0)(start) != ' '
+        || a(0)(end-1) != ' '
+        || a(2)(start) != ' '
+        || a(2)(end-1) != ' ') {
+        throwBE("Corners of state box not empty")
+    }
+
+    // NEWS order, strings
+    val str_surrs = List(
+      ("Top", a(0).slice(start,end) filter (_ != ' ')), // TODO: change to partition or something
+      ("Right", a(1)(end-1).toString),
+      ("Left", a(1)(start).toString),
+      ("Bottom", a(2).slice(start,end)  filter (_ != ' '))
+    )
+
+    val surrs = str_surrs map (_ match {
+        case (_, "#") => Blocked
+        case (_, "_") => Open
+        case (_, "*") => Anything
+        case (dir, s) => throw new CharBlockError(s"Invalid Surrounding `$s' for $dir side", start, end)
+      })
+
+    (Surroundings(surrs(0), surrs(1), surrs(2), surrs(3)), State(state_str))
+  }
+
+  def parseOutState(a: Seq[String],
+                    start: Int,
+                    end: Int,
+                    surrs: Surroundings) : (MoveDirection, State) = {
+    def throwBE(m : String) = throw new CharBlockError(m, start, end)
+  
+    //println(s"parseOutState: $a")
+    assert(a.length == 3)
+
+    val sym = "[#_*]"
+    val id = "[^#_*/\\-> |]*"
+    // >0 spaces, then either an #, o, or *, or a state name, then >0 spaces
+    val topBotRegex = s"""^ +(${id}|${sym}) +$$""".r
+    // id or sym, id or spaces, id or sym
+    val midRegex = s"""^(${id}|${sym})(${id}| +)(${id}|${sym})$$""".r
+
+    val tmo = topBotRegex findPrefixMatchOf a(0).slice(start,end)
+    val mmo = midRegex findPrefixMatchOf a(1).slice(start,end)
+    val bmo = topBotRegex findPrefixMatchOf a(2).slice(start,end)
+
+    // Unwrap options and make sure the regex was matched
+    val tm = tmo getOrElse throwBE("Top row malformed")
+    val mm = mmo getOrElse throwBE("Middle row malformed")
+    val bm = bmo getOrElse throwBE("Bottom row malformed")
+
+    // Verify column alignment
+    if (tm.start(1) < mm.start(2) || tm.end(1) > mm.end(2)) {
+      throwBE("top and middle not aligned")
+    }
+    if (bm.start(1) < mm.start(2) || bm.end(1) > mm.end(2)) {
+      throwBE("middle and bottom not aligned")
+    }
+    
+    // Strings that are in the north, east, west, south, and center positions
+    val matches = List(tm.group(1), mm.group(3), mm.group(1), bm.group(1), mm.group(2))
+
+    println("matches are " +  matches)
```

Comment:
This seems like it shouldn't always print - perhaps only if `verbose` mode were enabled (or something)